platforms/core-configuration/configuration-problems-base/build.gradle.kts

```diff
@@ -35,26 +35,29 @@ dependencies {
 
 tasks.processResources {
     from(zipTree(configurationCacheReportPath.elements.map { it.first().asFile })) {
-        into("org/gradle/internal/cc/impl/problems")
+        into("org/gradle/internal/configuration/problems")
         exclude("META-INF/**")
     }
 }
 
 dependencies {
     api(projects.baseServices)
-    api(projects.stdlibJavaExtensions)
-    api(projects.logging)
-    api(projects.problemsApi)
     api(projects.buildOption)
     api(projects.concurrent)
+    api(projects.coreApi)
     api(projects.fileTemp)
+    api(projects.logging)
     api(projects.loggingApi)
+    api(projects.messaging)
+    api(projects.problemsApi)
+    api(projects.stdlibJavaExtensions)
 
     api(libs.kotlinStdlib)
 
     implementation(libs.groovyJson)
+    implementation(libs.guava)
+
+    implementation(projects.core)
```

Comment:
❓  This is a big dependency, why do we need it now?

Reply:
for BuildNameProvider -> SettingsInternal